src/common/net.cc

```diff
@@ -104,21 +104,21 @@ Address::port() const {
 }
 
 void
-Address::init(std::string addr) {
-    auto pos = addr.find(':');
+Address::init(const std::string& addr) {
+    const auto pos = addr.find(':');
 
     if (pos == std::string::npos)
         throw std::invalid_argument("Invalid address");
 
-    std::string host = addr.substr(0, pos);
-    char *end;
+    host_ = addr.substr(0, pos);
 
+    char *end;
     const std::string portPart = addr.substr(pos + 1);
+    if (portPart.empty())
+        throw std::invalid_argument("Invalid port");
     long port = strtol(portPart.c_str(), &end, 10);
-    if (*end != 0 || port > Port::max())
+    if (*end != 0 || port < Port::min() || port > Port::max())
         throw std::invalid_argument("Invalid port");
-
-    host_ = std::move(host);
     port_ = port;
```

Comment:
You might want to make this "port_ = static_cast<uint16_t>(port);".  "port_" internally is uint16_t, and "port" is long.  You are doing proper range checking, but I can see a compiler throwing a warning about shoving a signed 32-bit int into an unsigned 16-bit int.

Reply:
@dennisjenkins75 You are totally right!